fifo.go

```diff
@@ -41,6 +41,11 @@ type fifo struct {
 
 var leakCheckWg *sync.WaitGroup
 
+var ReadWrOnlyFifoError = errors.New("reading from write-only fifo")
```

Comment:
var ()

Reply:
Suggestion, unless there is convention, can we either add From/Fr/From to the names or remove the cause so the variable name makes more sense? 

Example suggestions: ErrRdFrmWRONLY or ErrWRONLY